js/state/siteUtil.js

```diff
@@ -84,7 +85,26 @@ module.exports.getSiteKey = function (siteDetail) {
 }
 
 /**
- * Checks if a siteDetail has the specified tag
+ * Calculate location for siteKey
+ *
+ * @param siteKey The site key to to be calculated
+ * @return {string|null}
+ */
+module.exports.getLocationFromSiteKey = function (siteKey) {
+  if (!siteKey) {
+    return null
+  }
+
+  const splitKey = siteKey.split('|', 2)
+  if (typeof splitKey[0] === 'string' && typeof splitKey[1] === 'string') {
+    return splitKey[0]
+  }
+  return null
+}
+
+/**
+ * Checks if a siteDetail has the specified tag.
+ * O(n) warning; this iterates through sites.
```

Comment:
I think this should either be removed or it should just access in O(1) to check via site key.